src/main/java/duke/parser/ParserDuke.java

```diff
@@ -0,0 +1,129 @@
+package duke.parser;
+
+import duke.listobjects.Deadline;
+import duke.listobjects.Event;
+import duke.listobjects.ListObject;
+import duke.listobjects.ToDo;
+import duke.storage.Storage;
+import duke.tasklist.TaskList;
+
+
+/**
+ * Represents the Parser of Duke which parses and interprets user commands
+ */
+public class ParserDuke {
+
+    private String command;
+
+    private final String LIST_CMD = "list";
+    private final String MARK_CMD = "mark ";
+    private final String UNMARK_CMD = "unmark ";
+    private final String DELETE_CMD = "delete ";
+    private final String TODO_CMD = "todo ";
+    private final String EVENT_CMD = "event ";
+    private final String DEADLINE_CMD = "deadline ";
+    private final String FILE_PATH = "src/main/java/duke/DukeTasks.txt";
+
+    /**
+     * Constructs a ParserDuke object with given command
+     * @param command String representing user command
+     */
+    public ParserDuke(String command) {
+        this.command = command;
+    }
+
+    /**
+     * Prints response for empty user command
+     */
+    public void respondToEmptyString() {
+        System.out.println("The folly of youth to speak with no words! Speak again, my friend!");
+    }
+
+
+    /**
+     * Reads the user command, initiates completion of the appropriate action and prints response
+     */
+    public void parseCommand() {
+        if (command.isEmpty()) {
+            respondToEmptyString();
+        } else {
+            TaskList listOfItems = Storage.readFromFile(FILE_PATH);
+
+            if (command.equals(LIST_CMD)) {
```

Comment:
If else is fine but maybe switch is more readable?

Reply:
Yes, that would be better. Thanks!

Reply:
Maybe I could additionally employ enum and a separate class to improve the readability of the code...